llvm/lib/Transforms/InstCombine/InstructionCombining.cpp

```diff
@@ -646,6 +646,11 @@ static Value *tryFactorization(BinaryOperator &I, const SimplifyQuery &SQ,
   Value *LHS = I.getOperand(0), *RHS = I.getOperand(1);
   Instruction::BinaryOps TopLevelOpcode = I.getOpcode();
 
+  // Treat a disjoint Or like an Add if the inner opcode is a Mul.
```

Comment:
If I'm reading this right, it should always be legal to treat the disjoint or as and add, and the inner opcode check is to ensure it can distribute?  i.e. it's basically a profitability check?  If so, can you adjust the comment to be a bit more clear about that?

Reply:
If we treat it like an Add unconditionally, then we miss distribution opportunities on the Or because Add doesn't distribute over the same operations that Or does.